### ISSUE TITLE
Unlcear documentation of Attribute.

### ISSUE THREAD

**Issue by sallner** [flyingcircusio/batou#89]
Working with attributes for configuration I stumbled over this line  https://github.com/flyingcircusio/batou/blob/0d1fca74a1c1d5203d04f94263492c43e9bd81d2/src/batou/component.py#L1037 

Reading the `__get__` method afterwards looks as if the default is actually converted by passing it to `__set__` https://github.com/flyingcircusio/batou/blob/0d1fca74a1c1d5203d04f94263492c43e9bd81d2/src/batou/component.py#L1072

Also, multiple places of use show a behaviour which indicate a conversion of the default, e.g. https://github.com/flyingcircusio/batou/blob/0d1fca74a1c1d5203d04f94263492c43e9bd81d2/src/batou/lib/supervisor.py#L186

If you could clarify the situation, I would also create a PR for the docstring chance.

**Comment by ctheune** [comment]
You are right. I'm actually wondering which way would be right. Would you rather like to see defaults parsed or not? Also, it looks like we might have a migration problem or need to consider doing something like: if it's already of the expected type - don't convert it, but specifically non-type conversions would be hard to do ...

**Comment by ctheune** [comment]
So, looking at this in detail we have an issue where we want a conversion function (not just a type factory) that helps crossing the IO boundary between config/secret files and regular Python data structures.

The conversion really should only be happening when handling strings from the config file. Data in Python should be using the proper types already.

I would really appreciate if we reduce the conversion to the explicit cases where we have a config string. To help with documentation it is worthwhile to allow specifying defaults as config strings but making this explicit:

* no implicit conversions any longer
* check docs and places where we mention the attributes (at least the Component class)
* add a `from_config_string` method to the Attribute  class
* review the locations where we have config strings and explicitly check for attributes that support from_config_string
* add `default_config` argument to the Attribute class in addition to `default` (but XOR their usage) that will be run through the conversion
